src/tests/e2etest/NetworkingTests.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using Microsoft.Azure.Devices;
+using System.Text.Json;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+
+namespace e2etesting
+{
+    public class NetworkingTests : E2eTest
+    {
+        string ComponentName = "Networking";
+        string PropertyName = "NetworkConfiguration";
+        public partial class Networking
+        {
+            public string InterfaceTypes { get; set; }
+            public string MacAddresses { get; set; }
+            public string IpAddresses { get; set; }
+            public string SubnetMasks { get; set; }
+            public string DefaultGateways { get; set; }
+            public string DnsServers { get; set; }
+            public string DhcpEnabled { get; set; }
+            public string Enabled { get; set; }
+            public string Connected { get; set; }
+        }
+
+        public void RegexTest_SingletonInterfaceProperties(string testString, string patternString)
+        {
+            string interfaceDelimiter = ";";
+            Regex regexPattern = new Regex(@patternString);
+            string[] result = Regex.Split(testString, interfaceDelimiter);
+            foreach (string piece in result)
+            {
+                Assert.True(regexPattern.IsMatch(piece));
+            }
+        }
+
+        public void RegexTest_Properties(string testString, string piecePatternString, string smallPatternString)
+        {
+            string interfaceDelimiter = ";";
+            string propertyDelimiter = ",";
+            Regex regexPattern = new Regex(@piecePatternString);
+            Regex smallRegexPattern = new Regex(@smallPatternString);
+            string[] result = Regex.Split(testString, interfaceDelimiter);
+            foreach (string piece in result)
+            {
+                Match match = regexPattern.Match(piece);
+                Assert.True(match.Success);    
+                string[] tokens = Regex.Split(match.Groups[1].Value, propertyDelimiter);
+                foreach(string token in tokens)
+                {
+                    Assert.True(smallRegexPattern.IsMatch(token));
+                }
+            }
+        }
+
+        [Test]
+        public void NetworkingTest_Get()
+        {
+            if ((GetTwin().ConnectionState == DeviceConnectionState.Disconnected) || (GetTwin().Status == DeviceStatus.Disabled))
```

Comment:
first conditional argument should use `GetNewTwin()` in order to have the most up-to-date twin

Reply:
actually i take this back, if this is a reported only module, i would just make use of `GetTwin`